publisher.py

```diff
@@ -370,8 +370,8 @@ def publish_image(self, target: Registry, repository: str, tag: str):
                     message = line['errorDetail']['message']
                     raise ImagePushError(message=message)
                 self.log.debug(line)
-        except TimeoutError as e:
-            raise ImagePushError(message=e.message)
+        except ReadTimeoutError as e:
```

Comment:
I think that you can make a regression and don't catch other timeout exceptions like connection timeout.